api/v1alpha4/cluster_types.go

```diff
@@ -190,7 +191,7 @@ func (v APIEndpoint) IsValid() bool {
 
 // String returns a formatted version HOST:PORT of this APIEndpoint.
 func (v APIEndpoint) String() string {
-	return fmt.Sprintf("%s:%d", v.Host, v.Port)
+	return net.JoinHostPort(v.Host, fmt.Sprint(v.Port))
```

Comment:
@jsturtevant I wonder if this removes the need for the workaround we added in https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/646/files#diff-f35553213bd879e86978f9137fd9537ee287bb258030fbc4e69d24cab328f4a4R9 🤔

Reply:
could be, at the time I think we thought the health check was only doing ipv4 calls: https://github.com/jsturtevant/cluster-api-provider-azure/blob/c05182f6e9871ce529ddd645ea0433bc262ea5eb/docs/book/src/topics/ipv6.md#known-limitations

Reply:
actually, I don't think it would. This only helps in the case where the API Endpoint host is a public IP. We're using the DNS name in CAPZ.